File: server/Models/VaultKept.cs
using System.ComponentModel.DataAnnotations;


namespace keepr.Models;

public class VaultKept : RepoItem<int>
{
  public int VaultKeepId { get; set; }
  [MaxLength(255)] public string Name { get; set; }
  [MaxLength(1000)] public string Description { get; set; }
  [Url, MaxLength(1000)] public string Img { get; set; }
  public int Views { get; set; }
  public int Kept { get; set; }
  [MaxLength(255)] public string CreatorId { get; set; }
  public Profile Creator { get; set; }

}

File: client/src/models/VaultKept.js
import { APIItem } from "./APIItem.js";
import { Profile } from "./Profile.js";

export class VaultKept extends APIItem{
  constructor(data){
    super(data)
    this.vaultKeepId = data.vaultKeepId
    this.name = data.name
    this.description = data.description
    this.img = data.img
    this.views = data.views
    this.kept = data.kept
    this.creatorId = data.creatorId
    this.creator = new Profile(data.creator)
  }
}